odoo/orm/fields.py

```diff
@@ -529,8 +529,13 @@ def get_depends(self, model: BaseModel) -> tuple[Iterable[str], Iterable[str]]:
             if self._depends_context is not None:
                 depends_context = self._depends_context
             else:
-                related_model = model.env[self.related_field.model_name]
-                depends, depends_context = self.related_field.get_depends(related_model)
+                depends_context = []
+                for field_name in self.related.split('.'):
+                    field = model._fields[field_name]
+                    depends_context.extend(field.get_depends(model)[1])
+                    if field.relational:
+                        model = model.env[field.comodel_name]
```

Comment:
You are no longer computing `depends`?
```suggestion
                depends_context = []
                comodel = model
                for field_name in self.related.split('.'):
                    field = model._fields[field_name]
                    depends, rel_depends_context = field.get_depends(model)
                    depends_context.extend(rel_depends_context)
                    if not field.relational:
                        break
                    comodel = comodel.env[field.comodel_name]
```

Reply:
which `depends` do you mean? We always use `[self.related]` as the `depends` of the return value.